addons/mail/static/src/model/store.js

```diff
@@ -19,10 +19,21 @@ export class Store extends Record {
         return this.recordByLocalId.get(localId);
     }
 
+    DO_NOT_FAIL(fn) {
+        let res;
+        try {
+            res = fn();
+        } catch (err) {
+            console.warn("Internal Store Error:");
+            console.warn(err);
+        }
+        return res;
+    }
+
     /** @param {() => any} fn */
     MAKE_UPDATE(fn) {
         this._.UPDATE++;
-        const res = fn();
+        const res = this.DO_NOT_FAIL(() => fn());
```

Comment:
wouldn't it be the same if it was forwarded without the arrow function in that case?
```suggestion
        const res = this.DO_NOT_FAIL(fn);
```